### DISCUSSION TITLE
when i use vim press key end or home  the vim inputed error char

### DISCUSSION THREAD

**Discussion by yjc-xc** [sourcelair/xterm.js#5170] [General]
exp : i use xterm to exec vim 1.txt 
then press key end 
﻿
then edit mode actived and  char F   input in to cursor and wrap new line .
﻿
I expect cursor goto end of the line .
﻿
press key home also
﻿
help !  help ! help! 
![image](https://github.com/user-attachments/assets/2929769a-2388-4231-afb8-ab78036b306b)

**Comment by jerch** [comment]
Cant repro here.

If you want someone to help you, plz provide more context info, e.g. how you bind to the OS terminal (PTY), which versions you use etc.

(My best guess here is, that you dont bind correctly to the OS terminal, because if this would be an error within xtermjs we would see a flood of issue reports.)

**Reply by yjc-xc** [reply]
i got key  from xterm  then encode to base64  send to java service  .the java service use jsch to send linux

**Reply by yjc-xc** [reply]
use term.onkey(e) method , send to java  content is e.key .onData method also

**Reply by yjc-xc** [reply]
i find  when nomal press end  the e.key is \x[F ,but vim is \xOF . i try to convert . got fail

**Reply by jerch** [reply]
Does jsch provide a TTY stream setting? If not, then it is most likely a normal pipe stream, thus meant to run on slave side of a PTY. For that to work properly, you have to create another PTY in your server component and route the IO the the slave side of that PTY. The master then goes into xtermjs.

If it provides a TTY setting, you should activate it. Then you can route the IO directly into xtermjs.

**Reply by yjc-xc** [reply]
yes  jsch provide tty channel . nano is right  . vim got fail. the java service route input contents to jsch.  front encode to base64 .back decode base64

**Reply by jerch** [reply]
Then jsch maybe does not correctly forward DECKPAM sequences to the backend? Because in DECKPAM [End] would send `SS3 F` (`\x1b0F`).

**Reply by jerch** [reply]
also plz use `onData` instead of `onKey`. `onData` already correctly translates to the needed PTY symbols.

**Reply by yjc-xc** [reply]
thank you !  i try to resolve
